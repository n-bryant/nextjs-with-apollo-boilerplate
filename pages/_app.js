import React from "react";
import App from "next/app";

import Page from "../src/components/Page";
import { ApolloProvider } from "react-apollo";
import CssBaseline from "@material-ui/core/CssBaseline";

import withData from "../lib/withData";
import NextNProgress from "../src/components/NextNProgress";

export class BoilerplateApp extends App {
  componentDidMount() {
    // Remove the server-side injected CSS.
    const jssStyles = document.querySelector("#jss-server-side");
    if (jssStyles) {
      jssStyles.parentElement.removeChild(jssStyles);
    }
  }

  static async getInitialProps({ Component, ctx }) {
    let pageProps = {};
    if (Component.getInitialProps) {
      pageProps = await Component.getInitialProps(ctx);
    }
    // expose query to the user
    pageProps.query = ctx.query;
    return { pageProps };
  }

  render() {
    const { Component, apollo, pageProps } = this.props;

    return (
      <React.Fragment>
        <CssBaseline />
        <NextNProgress />
        <ApolloProvider client={apollo}>
          <Page>
            <Component {...pageProps} />
          </Page>
        </ApolloProvider>
      </React.Fragment>
    );
  }
}

export default withData(BoilerplateApp);
