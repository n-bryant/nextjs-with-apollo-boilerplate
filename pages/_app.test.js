import { BoilerplateApp } from "./_app";

describe("BoilerplateApp", () => {
  it("should render successfully", () => {
    const props = {
      Component: () => <div></div>,
      apollo: jest.fn(),
      pageProps: {}
    };

    const wrapper = shallow(<BoilerplateApp {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
