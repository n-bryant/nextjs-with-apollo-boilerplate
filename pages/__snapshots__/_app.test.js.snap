// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoilerplateApp should render successfully 1`] = `
<Fragment>
  <CssBaseline />
  <NextNProgress
    color="red"
    height={3}
    startPosition={0.3}
    stopDelayMs={200}
  />
  <ApolloProvider
    client={[MockFunction]}
  >
    <WithStyles(Page)
      classes={Object {}}
    >
      <Component />
    </WithStyles(Page)>
  </ApolloProvider>
</Fragment>
`;
